include/client/job/job_queue.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef JOB_QUEUE_H
+#define JOB_QUEUE_H
+
+#include "job.hpp"
+
+#include <queue>
+#include <condition_variable>
+#include <memory>
+#include <mutex>
+#include <thread>
+
+class JobQueue {
+
+public:
+    JobQueue();
+
+    static void push(Job &t_job);
+    static bool pop_try(Job &t_job);
+    static void push_res(Job &t_job);
+    static bool pop_res(Job &t_job);
+    static bool empty();
+
+private:
+
+    inline static std::mutex m_job_mutex;
+    inline static std::mutex m_res_mutex;
+    inline static std::queue<std::shared_ptr<Job>> m_queue;
+    inline static std::queue<std::shared_ptr<Job>> m_res;
+    inline static std::condition_variable m_datacond;  
```

Comment:
is there a reason why you made this static? 

I can see in your code that you might want to have 2 queues.  Why not instantiate the `JobQueue` twice, one for each of your queues?

```
JobQueue job_queue();
JobQueue res_queue();
```